Fixme: SA1019: http.CloseNotifier is deprecated

Tracking issue for some remaining linting issues to be fixed after https://github.com/moby/moby/pull/39668 is merged.

```
pkg/authorization/response.go:18:2:                   SA1019: http.CloseNotifier is deprecated: the CloseNotifier interface predates Go's context package. New code should use Request.Context instead.  (staticcheck)
pkg/authorization/response.go:160:30:                 SA1019: http.CloseNotifier is deprecated: the CloseNotifier interface predates Go's context package. New code should use Request.Context instead.  (staticcheck)
```

These warnings are currently suppressed in `hack/validate/golangci-lint.yml` (after https://github.com/moby/moby/pull/39668 is merged), so when this is fixed, those lines should be removed

--- a/hack/validate/golangci-lint.yml
+++ b/hack/validate/golangci-lint.yml
@@ -62,10 +62,6 @@ issues:
     - text: "SA1019: hdr.Xattrs is deprecated: Use PAXRecords instead"
       linters:
         - staticcheck
-    # FIXME temporarily suppress these. See #39929
-    - text: "SA1019: http.CloseNotifier is deprecated"
-      linters:
-        - staticcheck
     # FIXME temporarily suppress these. See #39926
     - text: "SA1019: httputil.NewClientConn is deprecated"
       linters:

--- a/pkg/authorization/response.go
+++ b/pkg/authorization/response.go
@@ -15,7 +15,6 @@ import (
 type ResponseModifier interface {
 	http.ResponseWriter
 	http.Flusher
-	http.CloseNotifier
 
 	// RawBody returns the current http content
 	RawBody() []byte
@@ -153,16 +152,6 @@ func (rm *responseModifier) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 		return nil, nil, fmt.Errorf("Internal response writer doesn't support the Hijacker interface")
 	}
 	return hijacker.Hijack()
-}
-
-// CloseNotify uses the internal close notify API of the wrapped http.ResponseWriter
-func (rm *responseModifier) CloseNotify() <-chan bool {
-	closeNotifier, ok := rm.rw.(http.CloseNotifier)
-	if !ok {
-		logrus.Error("Internal response writer doesn't support the CloseNotifier interface")
-		return nil
-	}
-	return closeNotifier.CloseNotify()
 }
 
 // Flush uses the internal flush API of the wrapped http.ResponseWriter

